src/passes/CoalesceLocals.cpp

```diff
@@ -440,7 +440,7 @@ void CoalesceLocals::applyIndices(std::vector<Index>& indices, Expression* root)
   auto numParams = getFunction()->getNumParams();
   Index newNumLocals = 0;
   for (auto index : indices) {
-    newNumLocals = std::max(newNumLocals, index + 1);
+    newNumLocals = std::max((Index::value_type)newNumLocals, index + 1);
```

Comment:
I don't understand why this is needed, which suggests I don't understand something important. Both `index` and `numNewLocals` are of type `Index`. Why doesn't this just work without a cast, then?